x/iro/types/bonding_curve.go

```diff
@@ -7,17 +7,49 @@ import (
 )
 
 /*
-with the following actions:
-  - SpotPrice(x) = M*x^N + C
-  - Cost(x, x1) = integral(x1) - integral(x)
-    The integral of y = m * x^N + c is (m / (N + 1)) * x^(N + 1) + c * x.
+A bonding curve is a mathematical function that defines the relationship between the price and supply of a token.
+The general form of a bonding curve is typically expressed as P = M * x^N + C, where:
+- P is the price
+- X is the supply
+- M is a multiplier that affects the curve's steepness
+- N is the exponent that determines the curve's shape
+- C is a constant that sets the initial price
+
+The implications of these parameters are significant:
+M (multiplier) controls the overall steepness of the curve. A higher M value results in a steeper price increase as supply grows, potentially leading to more rapid value appreciation but also higher volatility.
+
+N (exponent) shapes the curve's trajectory. When N > 1, the curve becomes convex, accelerating price growth at higher supply levels, which can create strong incentives for early adoption. When 0 < N < 1, the curve is concave, slowing price growth as supply increases, which can promote more stable long-term growth.
+
+C (constant) sets the starting price when supply is zero, effectively establishing a price floor and influencing the token's initial accessibility.
 */
 
 const (
-	MaxNValue    = 10
-	MaxPrecision = 2
+	MaxNValue     = 2 // Maximum allowed value for the N parameter
```

Comment:
why we changed to 2?

Reply:
reviewing real world scenarios, can't see how it can be usable
even N=2 is probably too high
the initial N=10 was out of the blue